install/swiftly-install.sh

```diff
@@ -356,20 +365,24 @@ USAGE:
 
 OPTIONS:
     -y, --disable-confirmation  Disable confirmation prompts.
-    --no-modify-profile         Do not attempt to modify the profile file to set environment 
+    -n, --no-modify-profile     Do not attempt to modify the profile file to set environment
                                 variables (e.g. PATH) on login.
-    --no-install-system-deps    Do not attempt to install Swift's required system dependencies.
-    --no-import-pgp-keys        Do not attempt to import Swift's PGP keys.
+    --no-install-system-deps    Do not attempt to install Swift's required system dependencies (LINUX ONLY)
+    --no-import-pgp-keys        Do not attempt to import Swift's PGP keys. (LINUX ONLY)
     -p, --platform <platform>   Specifies which platform's toolchains swiftly will download. If
                                 unspecified, the platform will be automatically detected. Available
                                 options are "ubuntu22.04", "ubuntu20.04", "ubuntu18.04", "rhel9", and
-                                "amazonlinux2".
-    --overwrite                 Overwrite the existing swiftly installation found at the configured
+                                "amazonlinux2". (LINUX ONLY)
+    -o, --overwrite             Overwrite the existing swiftly installation found at the configured
                                 SWIFTLY_HOME, if any. If this option is unspecified and an existing
                                 installation is found, the swiftly executable will be updated, but
                                 the rest of the installation will not be modified.
     -h, --help                  Prints help information.
-    --version                   Prints version information.
+    -v, --version               Prints version information.
+
+NOTES:
+    macOS only works with the short options (e.g. -v and not --version).
```

Comment:
Is this a limitation in macOS bash?

Reply:
Also curious about this

Reply:
> GNU `getopt` is possible to install on macOS, so we could attempt to detect it or require it, but I'm not sure if its use is widespread enough to warrant either of those.

I tried seeing if we could ask users to install a GNU getopt on macOS easily, but even the homebrew version doesn't support what is needed by the installer script. Moving the installer script to Swift should hopefully give us much better, and more user-friendly options.